utils/node_local_test.py

```diff
@@ -3014,6 +3014,148 @@ def test_cont_clone(self):
+        dirname = join(path, 'test_dir/1d1/2d3/')
+        os.mkdir(dirname)
+        fname = join(dirname, 'f7')
+        with open(fname, 'w') as fd:
+            fd.write('test7')
+
+        dirname = join(path, 'test_dir/1d2/2d4/')
+        os.mkdir(dirname)
+        fname = join(dirname, 'f8')
+        with open(fname, 'w') as fd:
+            fd.write('test8')
+        dirname = join(path, 'test_dir/1d2/2d5/')
+        os.mkdir(dirname)
+        fname = join(dirname, 'f9')
+        with open(fname, 'w') as fd:
+            fd.write('test9')
+        dirname = join(path, 'test_dir/1d2/2d6/')
+        os.mkdir(dirname)
+        fname = join(dirname, 'f10')
+        with open(fname, 'w') as fd:
+            fd.write('test10')
+
+        dirname = join(path, 'test_dir/1d3/2d7/')
+        os.mkdir(dirname)
+        fname = join(dirname, 'f11')
+        with open(fname, 'w') as fd:
+            fd.write('test11')
+        dirname = join(path, 'test_dir/1d3/2d8/')
+        os.mkdir(dirname)
+        fname = join(dirname, 'f12')
+        with open(fname, 'w') as fd:
+            fd.write('test12')
+        dirname = join(path, 'test_dir/1d3/2d9/')
+        os.mkdir(dirname)
+        fname = join(dirname, 'f13')
+        with open(fname, 'w') as fd:
+            fd.write('test13')
+
+        sys.path.append(join(self.conf['PREFIX'], 'lib64'))
+        daos_mw_fi = join(self.conf['PREFIX'], 'lib/daos/TESTING/tests/', 'daos_mw_fi')
+        cmd_env = get_base_env()
+        cmd_env['DAOS_AGENT_DRPC_DIR'] = self.conf.agent_dir
+
+        dir1 = join(path, 'test_dir/')
+        dir_list = os.listdir(dir1)
+        nr_entries = len(dir_list)
+        if nr_entries != 4:
+            raise NLTestFail('Wrong number of entries')
+
+        cmd = [daos_mw_fi, pool, cont, "punch_entry", "/test_dir/1d1/"]
+        print(cmd)
+        rc = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                            env=cmd_env, check=False, cwd=self.dfuse.dir)
+        if rc.stderr != b'':
+            print('Stderr from command')
+            print(rc.stderr.decode('utf-8').strip())
+        if rc.stdout != b'':
+            print(rc.stdout.decode('utf-8').strip())
+
+        dir_list = os.listdir(dir1)
+        nr_entries = len(dir_list)
+        if nr_entries != 3:
+            raise NLTestFail('Wrong number of entries')
+
+        cmd = [daos_mw_fi, pool, cont, "punch_entry", "/test_dir"]
+        print(cmd)
+        rc = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                            env=cmd_env, check=False, cwd=self.dfuse.dir)
+        if rc.stderr != b'':
+            print('Stderr from command')
+            print(rc.stderr.decode('utf-8').strip())
+        if rc.stdout != b'':
+            print(rc.stdout.decode('utf-8').strip())
+
+        cmd = ['fs', 'check', pool, cont, '--flags', 'print,relink', '--dir-name', 'lf']
+        rc = run_daos_cmd(self.conf, cmd)
+        print(rc)
+
```

Comment:
You could probably call the command here again and check/verify that it doesn't find anything a second time, the output there should be deterministic so you could verify it.

Reply:
yes, but trying to get it to pass in CI first